orderer/common/cluster/clusterservice.go

```diff
@@ -149,12 +154,29 @@ func (s *ClusterService) VerifyAuthRequest(stream orderer.ClusterNodeService_Ste
 		return nil, errors.Errorf("channel %s not found in config", authReq.Channel)
 	}
 
-	identity := membership.MemberMapping[authReq.FromId]
-	if identity == nil {
+	fromIdentity := membership.MemberMapping[authReq.FromId]
+	if fromIdentity == nil {
 		return nil, errors.Errorf("node %d is not member of channel %s", authReq.FromId, authReq.Channel)
 	}
 
-	err = VerifySignature(identity, SHA256Digest(msg), authReq.Signature)
+	toIdentity := membership.MemberMapping[authReq.ToId]
+	if toIdentity == nil {
+		return nil, errors.Errorf("node %d is not member of channel %s", authReq.ToId, authReq.Channel)
+	}
+
+	identityPEMBlock, _ := pem.Decode(s.NodeIdentity)
+	if identityPEMBlock == nil {
+		return nil, errors.Errorf("cert decoding failed")
+	}
+	toIdentityPEMblock, _ := pem.Decode(toIdentity)
+	if toIdentityPEMblock == nil {
+		return nil, errors.Errorf("cert decoding failed")
+	}
+	if err := crypto.CertificatesWithSamePublicKey(identityPEMBlock.Bytes, toIdentityPEMblock.Bytes); err != nil {
```

Comment:
Let's just compare bit-wise between `membership.MemberMapping[authReq.ToId]` and our identity. This PK based comparison is too much.